internal/statistics/state/state.go

```diff
@@ -0,0 +1,65 @@
+package state
+
+import (
+	"github.com/Izzette/go-safeconcurrency/eventloop/snapshot"
+	"github.com/benbjohnson/immutable"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+// State holds the statistics for the pods and image pulls.
+// State is immutable, all the methods return a new instance of the struct.
+// Do not lose track of the returned instance, it should be assigned to the containing structure.
+type State struct {
+	podStatistics       *PodStatistics
```

Comment:
PodStatistics could feasibly be squashed into `State`.  Alternatively two event loops could be used for pod statistics and image pull statistics, if it would simplify state management (there's no performance issue here, the events are _extremely_ fast thanks to CoW and zerolog)